app/models/missing_link_finder.rb

```diff
@@ -7,40 +15,63 @@ class MissingLinkFinder
 
   def initialize(redirection)
     @redirection = redirection
+    @session = Capybara::Session.new(:mechanize)
   end
 
   def run
-    begin
-      response = fetch(redirection.url)
-      if response.is_a?(Net::HTTPSuccess)
-        missing = missing_links(response.body)
-        if missing.empty?
-          { status: :good }
-        else
-          { status: :missing_links, missing: missing }
-        end
+    session.visit(redirection.url)
+    if session.status_code.in?(200..299)
+      if missing_links.empty?
+        {status: :good}
       else
-        { status: :offline }
+        {status: :missing_links, missing: missing_links}
       end
-    rescue SocketError
-      { status: :offline }
-    rescue StandardError => error
-      { status: :error, error: "#{error.class}: #{error.message}" }
+    else
+      {status: :offline}
+    end
+  rescue SocketError
+    {status: :offline}
+  rescue => error
+    if error.message.include?("ERR_NAME_NOT_RESOLVED")
+      {status: :offline}
+    elsif error.message.include?("Net::HTTPNotFound")
+      {status: :not_found}
+    else
+      {status: :error, error: "#{error.class}: #{error.message}"}
     end
   end
 
   private
 
-  attr_reader :redirection
+  attr_reader :redirection, :session
 
-  def missing_links(html)
-    if html
-      missing_links = []
-      missing_links << "next" if !html.match?(next_link)
-      missing_links << "prev" if !html.match?(prev_link)
+  def missing_links
+    found_next = has_next_link?
+    found_previous = has_prev_link?
+    if !found_next || !found_previous
```

Comment:
I'm finding the negation and or a little weird to read (not this or not that). I wonder if instead of `has_<some>_link` the methods were `missing_<some>_link?` instead? Alternatively, we could return early:

```suggestion
  return if has_next_link? && has_prev_link?
```